src/core/draw_tools/atoms/combinedAtom.ts

```diff
@@ -95,9 +100,17 @@ export const combinedAtom = createAtom(
       actions.push(temporaryGeometryAtom.setFeatures(features, indexes));
     });
 
-    onChange('drawnGeometryAtom', (featureCollection) =>
-      (state.drawnGeometryAtom ?? []).forEach((cb) => cb(featureCollection)),
-    );
+    onChange('drawnGeometryAtom', (featureCollection, prevCollection) => {
+      if (selectCurrentGeometryWasRequested) {
+        actions.push(selIndexesForCurrentGeometryAtom.set(false));
+        const indexes: number[] = [];
+        for (let i = 0; i < featureCollection.features.length; i++) {
```

Comment:
you can use https://lodash.com/docs/4.17.15#times and make it bit more compact

Reply:
or [...Array(featureCollection.features.length).keys()]

Reply:
you guys clearly don't like vanilla js :D